File: scala/scala-impl/src/org/jetbrains/plugins/scala/project/settings/ScalaCompilerSettingsProfile.java
package org.jetbrains.plugins.scala.project.settings;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

/**
 * @author Pavel Fatin
 */
// TODO This class is needed for the "imported" ScalaCompilerConfigurationPanel.
// TODO It's better to replace it with immutable case classes later.
public class ScalaCompilerSettingsProfile {
  private String myName;
  private List<String> myModuleNames = new ArrayList<String>();
  private ScalaCompilerSettings mySettings = new ScalaCompilerSettings();

  public ScalaCompilerSettingsProfile(String name) {
    myName = name;
  }

  public String getName() {
    return myName;
  }

  public void initFrom(ScalaCompilerSettingsProfile profile) {
    myName = profile.getName();
    mySettings = profile.getSettings();
    myModuleNames = new ArrayList<String>(profile.getModuleNames());
  }

  public List<String> getModuleNames() {
    return Collections.unmodifiableList(myModuleNames);
  }

  public void addModuleName(String name) {
    myModuleNames.add(name);
  }

  public void removeModuleName(String name) {
    myModuleNames.remove(name);
  }

  public ScalaCompilerSettings getSettings() {
    return mySettings;
  }

  public void setSettings(ScalaCompilerSettings settigns) {
    mySettings = settigns;
  }

  @Override
  public String toString() {
    return myName;
  }
}

File: scala/scala-impl/src/org/jetbrains/plugins/scala/project/settings/ScalaCompilerSettingsProfile.scala
package org.jetbrains.plugins.scala.project.settings

import java.util
import java.util.Collections

/**
 * @author Pavel Fatin
 */
// TODO This class is needed for the "imported" ScalaCompilerConfigurationPanel.
// TODO It's better to replace it with immutable case classes later.
class ScalaCompilerSettingsProfile(name: String) {
  private var myName: String = name
  private var myModuleNames = new util.ArrayList[String]
  private var mySettings = new ScalaCompilerSettings

  def getName: String = myName

  def initFrom(profile: ScalaCompilerSettingsProfile): Unit = {
    ScalaCompilerConfiguration.incModificationCount()
    myName = profile.getName
    mySettings = profile.getSettings
    myModuleNames = new util.ArrayList[String](profile.getModuleNames)
  }

  def getModuleNames: util.List[String] = Collections.unmodifiableList(myModuleNames)

  def addModuleName(name: String): Unit = {
    ScalaCompilerConfiguration.incModificationCount()
    myModuleNames.add(name)
  }

  def removeModuleName(name: String): Unit = {
    ScalaCompilerConfiguration.incModificationCount()
    myModuleNames.remove(name)
  }

  def getSettings: ScalaCompilerSettings = mySettings

  def setSettings(settigns: ScalaCompilerSettings): Unit = {
    ScalaCompilerConfiguration.incModificationCount()
    mySettings = settigns
  }

  override def toString: String = myName
}
